### ISSUE TITLE
@extend cannot be used from within @media

### ISSUE THREAD

**Issue by blainehansen** [zessx/sass-flexbox#3]
Since almost the entire library uses `@extend` from within mixins, the mixins fail when called from within media directives.

I'd like to suggest a reversal of the current strategy, where the mixins call `@extend`. Instead, the `@extend`s should call the mixins.

So from this:

```scss
%display-flex {
    display: -webkit-box;
    display: -moz-box;
    display: -webkit-flexbox;
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
}
@mixin display-flex() {
    @extend %display-flex;
}
```

to this:

```scss
@mixin display-flex() {
    display: -webkit-box;
    display: -moz-box;
    display: -webkit-flexbox;
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
}
%display-flex {
    @include display-flex();
}
```

**Comment by zessx** [comment]
Althought I would recommand to use Autoprefixer instead of Sass-flexbox, I think this is a really good idea, as it does not even change users' habits.  
I'll take a little time to think about eventual regressions.

**Comment by zessx** [comment]
Fixed with 7aa9e068693dbae9ea148bf28d0899ea65384e68

**Comment by zessx** [comment]
Fixed with 7aa9e068693dbae9ea148bf28d0899ea65384e68
